platform.py

```
@@ -19,234 +19,477 @@
+        ]
 
-        if "arduino" in frameworks:
-            self.packages["framework-arduinoespressif32"]["optional"] = False
-            self.packages["framework-arduinoespressif32-libs"]["optional"] = False
-            # use branch master
-            URL = "https://raw.githubusercontent.com/espressif/arduino-esp32/master/package/package_esp32_index.template.json"
-            packjdata = requests.get(URL).json()
-            dyn_lib_url = packjdata['packages'][0]['tools'][0]['systems'][0]['url']
-            self.packages["framework-arduinoespressif32-libs"]["version"] = dyn_lib_url
+        try:
+            with open(os.devnull, 'w') as devnull:
```

Ruff (0.11.9)

Local variable `devnull` is assigned to but never used

Remove assignment to unused variable `devnull`

(F841)